collector/textfile.go

```diff
@@ -237,12 +237,16 @@ fileLoop:
 		}
 		var parser expfmt.TextParser
 		parsedFamilies, err := parser.TextToMetricFamilies(carriageReturnFilteringReader{r: file})
-		file.Close()
 		if err != nil {
 			log.Errorf("Error parsing %q: %v", path, err)
 			error = 1.0
 			continue
 		}
+		err = file.Close()
```

Comment:
Is this correct? Looks like if err != nil above at line 240, the file wont get closed